src/nunavut/__init__.py

```diff
@@ -291,9 +306,16 @@ def __hash__(self) -> int:
     # +-----------------------------------------------------------------------+
     # | PRIVATE
     # +-----------------------------------------------------------------------+
-    def _add_data_type(self, dsdl_type: pydsdl.CompositeType, extension: typing.Optional[str]) -> None:
+    def _add_data_type(self, dsdl_type: pydsdl.CompositeType, extension: typing.Optional[str], snake_case: bool) \
```

Comment:
This is *extremely* hacky. I'm not really comfortable enough with the codebase to be confident in coming up with a proper idiomatic solution, so if anyone has any ideas, I'm all ears.

For reference, the problem I'm trying to solve:

With the default naming of files, I'll end up with a structure like `uavcan::node::Health_1_0::Health_1_0`, and I *want*
a structure like `uavcan::node::Health_1_0`, without the intermediate module. In order to preserve handy naming of types, I want to rename the intermediate module that contains the type, and the most idiomatic (in Rust anyways) way I can think of is changing the file name to snake case.

Reply:
This seems okay and is also exactly the way I handled it in Python.

Reply:
I'm working on a big refactor of the Language and Environment objects because they sort of blew up on me when I tried to add the concept of "using language features" (e.g. `{% ifuses "std_variant" %}...`). This seems co-morbid in that the snake casing probably wants to be shared logic with template filters and other uses.In general, I'd like to get this PR in sooner rather then later which should be fine since we have an "experimental language" flag on by default so users don't see half-baked support like C++ or Rust until we flip that switch. Can we find a good iteration point where we get the tests to run green and we can merge your work with my refactor?

Reply:
Seems like a good idea, but I'm also not entirely sure what you mean in terms of how we manage branches/merging.

The only things I need to get done are a proper solution to snake-casing everything, and the crate generation, basically all the type generation work is done excepting some clean up. Are you suggesting I pull in your refactor now and work with those code changes, or get what I'm doing done and then pull in your refactor?

Reply:
I'm just saying, don't be afraid to iterate in smaller chunks to keep your work from rotting. As long as the build passes and your changes do no harm it's okay to merge them even if incomplete because of the experimental languages flag.